llvm/lib/CodeGen/SelectionDAG/SelectionDAG.cpp

```diff
@@ -954,6 +954,12 @@ static void AddNodeIDCustom(FoldingSetNodeID &ID, const SDNode *N) {
       ID.AddInteger(M);
     break;
   }
+  case ISD::ADDRSPACECAST: {
+    const AddrSpaceCastSDNode *ASC = cast<AddrSpaceCastSDNode>(N);
```

Comment:
Unrelated change

Reply:
I've spun this out into https://github.com/llvm/llvm-project/pull/122912